screwdriver/screwdriver_test.go

```diff
@@ -0,0 +1,288 @@
+		}))
+
+		gotAPI, err := New(server.URL, testToken)
+		api, ok := gotAPI.(*sdAPI)
+		assert.True(t, ok)
+		assert.Nil(t, err)
+		assert.Equal(t, testToken, api.UserToken)
+		assert.Equal(t, server.URL, api.APIURL)
+		assert.Equal(t, testJWT, api.SDJWT)
+		assert.Equal(t, testJWT, api.JWT())
+	})
+
+	t.Run("failure by invalid JSON", func(t *testing.T) {
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(200)
+			w.Header().Set("Content-Type", "application/json")
+
+			testBody := fmt.Sprint(`{`)
+			fmt.Fprintln(w, testBody)
+		}))
+
+		testToken := "token"
+		_, err := New(server.URL, testToken)
+		assert.NotNil(t, err)
+
+		testMsg := "failed to parse JWT response:"
+		assert.Contains(t, err.Error(), testMsg)
+
+	})
+
+	t.Run("failure by status", func(t *testing.T) {
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(500)
+		}))
+
+		testToken := "token"
+		_, err := New(server.URL, testToken)
+		assert.NotNil(t, err)
+
+		testMsg := "failed to get JWT: StatusCode 500"
+		assert.Equal(t, testMsg, err.Error())
+	})
+
+	t.Run("failure by makeURL", func(t *testing.T) {
+		testURL := "http://example.com:yyy"
+		testMsg := "failed to make request url:"
+		testToken := "token"
+		_, err := New(testURL, testToken)
+		assert.Contains(t, err.Error(), testMsg)
+	})
+
+	t.Run("failure by sd.request", func(t *testing.T) {
+		testURL := "http://localhost"
+		testMsg := "connection refused"
+		testToken := "testToken"
+		_, err := New(testURL, testToken)
+		assert.Contains(t, err.Error(), testMsg)
+	})
+}
+
+func TestJob(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			wantContentType := "application/json"
+			validateHeader(t, "Content-Type", wantContentType, r)
+			wantAuthBearer := "Bearer jwt"
+			validateHeader(t, "Authorization", wantAuthBearer, r)
+
+			w.WriteHeader(200)
+			w.Header().Set("Content-Type", "application/json")
+
+			testJSON, err := ioutil.ReadFile(path.Join(testDir, "validatedSuccess.json"))
+			assert.Nil(t, err)
+			fmt.Fprintln(w, string(testJSON))
+		}))
+
+		testAPI := sdAPI{
+			HTTPClient: http.DefaultClient,
+			UserToken:  "dummy",
+			APIURL:     server.URL,
+			SDJWT:      "jwt",
+		}
+
+		testJob := Job{
+			Steps: []Step{
+				{
+					Name:    "install",
+					Command: "echo install",
+				},
+				{
+					Name:    "publish",
+					Command: "echo publish",
+				},
+			},
+			Environment: map[string]string{
+				"TEST_ENV": "hoge",
+			},
+			Image: "alpine",
+		}
+
+		gotJob, err := testAPI.Job("main", path.Join(testDir, "screwdriver.yaml"))
+		assert.Nil(t, err)
+		assert.Equal(t, testJob, gotJob)
+	})
+
+	t.Run("failure by making URL", func(t *testing.T) {
+		testAPI := sdAPI{
+			HTTPClient: http.DefaultClient,
+			UserToken:  "dummy",
+			APIURL:     string(0x7f),
```

Comment:
It is hard to understand.
Is it better to change it like [this](https://github.com/screwdriver-cd/sd-local/blob/5b14f95ce778ab0b775b48cce50fe25bdc979878/screwdriver/screwdriver_test.go#L85-L89)?

Reply:
Fixed.